src/bundles.ts

```diff
@@ -249,3 +261,32 @@ export async function createBundleFromFolders(options: CreateBundleFromFoldersOp
     fileIgnores,
   };
 }
+
+export function convertLanguagesToExtensions(languages: string[]): string[] {
+  const langDictionary = {
+    csharp: 'c#',
+    cpp: 'c++',
+  };
+  const extensions =
+    languages
+      .map(language => {
+        if (langDictionary[language]) {
+          return langMap.extensions(langDictionary[language]);
+        }
+        return langMap.extensions(language);
+      })
+      .flat()
+      .map(language => `.${language}`) || [];
+
+  return extensions;
+}
+
+function getAllowedExtensions(systemExtensions: string[], customerExtensions: string[]): string[] {
+  const allowedExtensions = customerExtensions.filter(ce => {
+    if (systemExtensions.includes(ce)) {
+      return true;
+    }
+    return false;
+  });
+  return allowedExtensions;
+}
```

Comment:
@Arvi3d Bundle's supported file extension is a list of 30 items.

This means that if we are using a third-party library, we must make sure that it has the same (or more, but not less) numbers of extensions that Bundle supports. Otherwise, certain scans might not happen because `getAllowedExtensions()` will exclude their file extension. 

Full list of extensions provided by Bundle through code-client:
```
[
      '.aspx',
      '.c',
      '.cc',
      '.cls',
      '.cpp',
      '.cs',
      '.cxx',
      '.ejs',
      '.erb',
      '.es',
      '.es6',
      '.go',
      '.h',
      '.haml',
      '.hpp',
      '.htm',
      '.html',
      '.hxx',
      '.java',
      '.js',
      '.jsx',
      '.php',
      '.phtml',
      '.py',
      '.rb',
      '.rhtml',
      '.slim',
      '.ts',
      '.tsx',
      '.vue',
    ];
   ```